packages/react-composites/src/composites/CallComposite/components/CallControls.tsx

```diff
@@ -89,41 +98,43 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
     [props.isMobile, props.options]
   );
 
-  const adapter = useAdapter();
-
   const localeStrings = useLocale();
 
   /* @conditional-compile-remove(DNS) */
   const [isDeepNoiseSuppressionOn, setDeepNoiseSuppressionOn] = useState<boolean>(false);
 
+  /* @conditional-compile-remove(DNS) */
+  const adapter = useAdapter();
   /* @conditional-compile-remove(DNS) */
   const startDeepNoiseSuppression = useCallback(async () => {
     await adapter.startNoiseSuppressionEffect();
   }, [adapter]);
 
+  /* @conditional-compile-remove(DNS) */
+  const deepNoiseSuppresionEffectsDependency = useSelector(getDeepNoiseSuppresionEffectsDependency);
+  /* @conditional-compile-remove(DNS) */
+  const deepNoiseSuppressionOnByDefault = useSelector(getDeepNoiseSuppresionIsOnByDefault);
   /* @conditional-compile-remove(DNS) */
   useEffect(() => {
-    if (
-      adapter.getState().onResolveDeepNoiseSuppressionDependency &&
-      adapter.getState().deepNoiseSuppressionOnByDefault
-    ) {
+    if (deepNoiseSuppresionEffectsDependency && deepNoiseSuppressionOnByDefault) {
       startDeepNoiseSuppression();
       setDeepNoiseSuppressionOn(true);
     }
-  }, [adapter, startDeepNoiseSuppression]);
+  }, [adapter, deepNoiseSuppresionEffectsDependency, deepNoiseSuppressionOnByDefault, startDeepNoiseSuppression]);
```

Comment:
it looks like adapter is not needed in the deps list anymore

Reply:
Thanks! I wonder if there's a lint rule to catch unnecessary dependencies... (I guess you might want certain deps to trigger re-renders even if they're not used in the useEffect so its not a default linting rule?)

Reply:
Updated

Reply:
https://github.com/Azure/communication-ui-library/pull/5239#discussion_r1788046855
it looks like `react-hooks/exhaustive-deps` should work like this...

Reply:
Good point, that rule doesn't seem to be working in this file at all... Looking into it / will file a bug